neonvm/controllers/metrics.go

```diff
@@ -24,8 +32,32 @@ func MakeReconcilerMetrics() ReconcilerMetrics {
 			},
 			[]string{"controller"},
 		),
+		vmCreationToRunnerCreationTime: prometheus.NewHistogram(
+			prometheus.HistogramOpts{
+				Name:    "vm_creation_to_runner_creation_duration_seconds",
+				Help:    "Time duration from VirtualMachine.CreationTimestamp to runner Pod.CreationTimestamp",
+				Buckets: buckets,
+			},
+		),
+		runnerCreationToVMRunningTime: prometheus.NewHistogram(
+			prometheus.HistogramOpts{
+				Name:    "runner_creation_to_vm_running_duration_seconds",
+				Help:    "Time duration from runner Pod.CreationTimestamp to the moment when VirtualMachine.Status.Phase becomes Running",
+				Buckets: buckets,
+			},
+		),
+		vmCreationToVMRunningTime: prometheus.NewHistogram(
+			prometheus.HistogramOpts{
+				Name:    "vm_creation_to_vm_running_duration_seconds",
+				Help:    "Time duration from VirtualMachine.CreationTimeStamp to the moment when VirtualMachine.Status.Phase becomes Running",
+				Buckets: buckets,
+			},
+		),
 	}
 	metrics.Registry.MustRegister(m.failing)
+	metrics.Registry.MustRegister(m.vmCreationToRunnerCreationTime)
+	metrics.Registry.MustRegister(m.runnerCreationToVMRunningTime)
+	metrics.Registry.MustRegister(m.vmCreationToVMRunningTime)
```

Comment:
I'd recommend using `util.RegisterMetric` here so that registering can be done at the same time as initializing (avoids the possibility of issues like #787)

Reply:
I tried to use that but the types are incompatible and had to type casting in runtime. See this: https://github.com/kubernetes-sigs/controller-runtime/blob/4000e996a202917ad7d40f02ed8a2079a9ce25e9/pkg/metrics/registry.go#L30

Reply:
Ugh, ok yeah fair. Maybe worthwhile to change `util.RegisterMetric` so that it takes `prometheus.Registerer` rather than `*prometheus.Registry`? I _think_ that'd allow it to work here

Reply:
Cool, I didn't know about the `prometheus.Registerer`. Actually, it's better to use the interface in the `util.RegisterMetric`. I'll send a separate PR for that.

Reply:
Opened and merged #792